tests/performance/ecommerce_hybrid_search/dataprep/generate_feed_query.py

```diff
@@ -0,0 +1,372 @@
+
+
+def clean_and_prepare_df(df: pd.DataFrame) -> pd.DataFrame:
+    df = df.drop_duplicates(subset=["title", "description"])
+    # Set index to id column (use .loc to avoid SettingWithCopyWarning)
+    df.loc[:, "id"] = df.index
+    # assert (
+    #     df["description"].str.strip().eq("").sum() == 0
+    # ), "Empty strings in descriptions"
+    assert df["id"].duplicated().sum() == 0, "Duplicated IDs found"
+    assert df.isnull().sum().sum() == 0, "NaN values found in DataFrame"
+    return df
+
+
+def save_df_to_vespa_format(df: pd.DataFrame, file_name: Path) -> None:
+    df = df.apply(
+        lambda row: {
+            "put": f"id:{VESPA_DOCTYPE}:{VESPA_NAMESPACE}::{row['category']}{row['id']}",
+            "fields": {
+                "id": row["id"],
+                "title": row["title"],
+                "category": row["category"],
+                "description": row["description"],
+                "price": row["price"],
+                "average_rating": row["average_rating"],
+                "embedding": row["embedding"].tolist(),
+            },
+        },
+        axis=1,
+    )
+    file_name = file_name.with_suffix(".json.zst")
+    df.to_json(file_name, orient="records", lines=True, compression="zstd")
+    logging.info(f"Data saved in Vespa format to {file_name}")
+
+
+def save_df_to_es_format(df: pd.DataFrame, file_name: Path) -> None:
+    file_name = file_name.with_suffix(".json.zst")
+    data = ""
+    for index, row in df.iterrows():
+        action = {"index": {"_index": ES_INDEX, "_id": str(row["id"])}}
+        data += json.dumps(action, ensure_ascii=True) + "\n"
+        doc_data = {
+            "title": row["title"],
+            "category": row["category"],
+            "description": row["description"],
+            "price": row["price"],
+            "average_rating": row["average_rating"],
+            "embedding": row["embedding"].tolist(),
+        }
+        data += json.dumps(doc_data, ensure_ascii=True) + "\n"
+    cctx = zstd.ZstdCompressor()
+    compressed_data = cctx.compress(data.encode("utf-8"))
+    with open(file_name, "wb") as f:
+        f.write(compressed_data)
+    logging.info(f"Data saved in Elasticsearch format to {file_name}")
+
+
+def title_to_query(title: str) -> str:
+    # If the title is more than 8 words, use the first 4 words.
+    # If it is less than 8 words, use the whole title.
+    return " ".join(title.split()[:8])
+
+
+def remove_quote_chars(query: str) -> str:
+    """Should remove both single and double quotes from the query."""
+    # Use regex compiled for performance
+    remove_double_quotes = re.compile(r'"')
+    remove_single_quotes = re.compile(r"'")
+    return remove_double_quotes.sub("", remove_single_quotes.sub("", query))
+
+
+def save_vespa_query_files_from_df(
+    df: pd.DataFrame,
+    query_save_path: Path,
+    num_queries: int,
+    query_type: QueryTypeEnum = QueryTypeEnum.WEAK_AND,
+) -> None:
+    """
+    # 1. weakAnd + bm25 ranking:

```

Comment:
I don't think we need lines 151 to 163, as the same information is found on lines 183 to 200 in the code.